tests/test_shaft.py

```diff
@@ -20,7 +20,8 @@ def test_generate_solid_02(self):
 
     def test_exception(self):
     	sh = Shaft("tests/test_datasets/parameters.prm")
-    	sh.generate_solid()
+    	with self.assertRaises(Exception):
+    		sh.generate_solid()
```

Comment:
please use spaces (4 spaces = 1 tab). It seems here you used a tab. Please check.